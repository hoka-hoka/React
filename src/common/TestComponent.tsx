import React, { Component } from 'react';

interface TestComponentProps {
  page: string;
  children: React.ReactNode;
}
interface TestComponentState {
  stateEvent: string;
}
export default class TestComponent extends Component<
  TestComponentProps,
  TestComponentState
> {
  constructor(props: TestComponentProps) {
    super(props);
    this.state = { stateEvent: '' };
  }

  onMouseEnter = () => {
    this.setState({ stateEvent: 'enter' });
  };

  onMouseLeave = () => {
    this.setState({ stateEvent: 'leave' });
  };

  render() {
    const { stateEvent } = this.state;
    const { page, children } = this.props;
    return (
      <a
        className={stateEvent}
        href={page}
        onMouseEnter={this.onMouseEnter}
        onMouseLeave={this.onMouseLeave}
      >
        {children}
      </a>
    );
  }
}
