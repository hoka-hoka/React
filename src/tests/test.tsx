import React from 'react';
import renderer from 'react-test-renderer';
import TestComponent from '../common/TestComponent';

test('Link', () => {
  const obj = {
    createAt: new Date(),
    id: Math.random(),
  };

  expect(obj).toMatchSnapshot();
});
