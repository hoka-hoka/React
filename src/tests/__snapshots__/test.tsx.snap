// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 1`] = `
Object {
  "createAt": 2021-06-19T09:46:01.014Z,
  "id": 0.06317999107421457,
}
`;
